src/lbaf/Utils/logging.py

```diff
@@ -13,87 +14,116 @@
+                )
+        # or create raw formatter to use at any level
+        else:
+            self._raw_formatter = Formatter(formats.get("prefix") + formats.get("message"))
+
     def format(self, record):
-        return self.formatters[record.levelno].format(record)
-
-def formatter(formatter_type: str, theme: Union[str, None] = None):
-    """Creates a formatter of the given type"""
-    if not formatter_type in FORMATTERS:
-        frmtrs = str.join(', ', FORMATTERS)
-        raise ValueError(f"Invalid formatter name. Supported are {frmtrs}")
-
-    if theme is not None and not theme in THEMES:
-        themes = str.join(', ', THEMES)
-        raise ValueError(f"Invalid thee name. Supported are {themes}")
-
-    def msgcolor(msg):
-        if not theme is None:
-            return black(msg) if theme == "light" else light_white(msg)
-        return msg
-
-    # 'extended' formatter
-    if formatter_type == "extended":
-        frmttr = {
-            logging.DEBUG: yellow("%(levelname)s [%(module)s.%(funcName)s()] ") + msgcolor("msg:[%(message)s]"),
-            logging.INFO: green("%(levelname)s [%(module)s.%(funcName)s()] ") + msgcolor("msg:[%(message)s]"),
-            logging.WARNING: cyan("%(levelname)s [%(module)s.%(funcName)s()] ") + msgcolor("msg:[%(message)s]"),
-            logging.ERROR: red("%(levelname)s [%(module)s.%(funcName)s()] ") + msgcolor("msg:[%(message)s]")
-        }
-    # 'basic' formatter
-    else:
-        frmttr = {
-            logging.DEBUG: yellow("[%(module)s] ") + msgcolor("%(message)s"),
-            logging.INFO: green("[%(module)s] ") + msgcolor("%(message)s"),
-            logging.WARNING: cyan("[%(module)s] ") + msgcolor("%(message)s"),
-            logging.ERROR: red("[%(module)s] ") + msgcolor("%(message)s")
-        }
-    return CustomFormatter(frmttr)
-
-d = {}
+        formatter = None
+        if self._colored:
+            formatter = self._color_formatters[
+                record.levelno
+                    if record.levelno in self.LOGGING_LEVEL_COLORS
+                    else logging.INFO
+            ]
+        else:
+            formatter = self._raw_formatter
+        return formatter.format(record)
+
+loggers = {}
 
 def get_logger(
         name: str = "root",
         level: Union[str, None] = "info",
         log_to_console: bool = True,
         log_to_file: Union[str, None] = None,
-        formatter_name: str = FORMATTER_BASIC,
-        theme: str = THEME_DARK
+        frmt: str = FORMAT_BASIC
 )-> Logger:
     """Return a new or an existing logger"""
+
     # return from cache if logger already created
-    logger = d.get(name, None)
-    if logger is not None:
-        return logger
+    # logger = logging.Logger.manager.loggerDict.get(name)
+    # if logger is not None:
+    #     return logger
+    if name in loggers:
+        return loggers[name]
+
+    print("Init logger " + name + "...")
```

Comment:
Write "Initializing" instead of "Init" and remove triple dots

Reply:
finally print removed